scene/resources/texture.cpp

```diff
@@ -1028,6 +1028,21 @@ void AtlasTexture::draw_rect(RID p_canvas_item, const Rect2 &p_rect, bool p_tile
 
 	atlas->draw_rect_region(p_canvas_item, dr, rc, p_modulate, p_transpose, p_normal_map);
 }
+
+Texture::RefineRectResult AtlasTexture::refine_rect_region(Rect2 &r_dst_rect, Rect2 &r_src_rect) const {
+	if (!atlas.is_valid()) {
+		return REFINE_RECT_RESULT_NO_DRAW;
+	}
+	Rect2 temp_rect = r_dst_rect;
+	Rect2 temp_src_rect = r_src_rect;
+
+	if (get_rect_region(temp_rect, temp_src_rect, r_dst_rect, r_src_rect)) {
+		return atlas->refine_rect_region(r_dst_rect, r_src_rect);
+	}
```

Comment:
If we want to avoid copying then this should be fine too (under assumption both `get_rect_region` and `refine_rect_region` are assigning the results at the end of their execution):
```suggestion
	if (get_rect_region(r_dst_rect, r_src_rect, r_dst_rect, r_src_rect)) {
		return atlas->refine_rect_region(r_dst_rect, r_src_rect);
	}
```
I've done it like that originally (pre opening my PR) but changed it for clarity. So if changing like suggested then probably adding some comment would be a good idea.

Reply:
I used the temps because imo it is dodgy territory using the same variable for both const ref and referenced return argument to the same function, because of aliasing. The function may assume that the const argument is unchanging (either the programmer of the function, or the compiler).

Using the temp enforces the const nature and prevents aliasing, and may well be compiled out if unneeded in this case (as `get_rect_region()` is known at compile time, even though virtual).

So imo changing this should be left to a future PR that changes the `get_rect_region()` function.